tree/treeplayer/src/TTreeFormula.cxx

```diff
@@ -4017,6 +4017,18 @@ T TTreeFormula::EvalInstance(Int_t instance, const char *stringStackArg[])
    Int_t pos  = 0;
    Int_t pos2 = 0;
    for (Int_t i=0; i<fNoper ; ++i) {
+      
+      if (pos >= kMAXFOUND) {
```

Comment:
I think we ought to try to find this error at parsing time rather than execution time to avoid spurrious test within the performance critical code.

Reply:
Sorry, I am not following, what do you mean with at parsing time? In a different part of this function?

Reply:
I see. But two questions:

why did the user report in JIRA the crash then exactly in TTreeFormula.cxx:3940 and not in TFormula itself? Are we sure that if we fix it in TFormula, it won't crash in TTreeFormula?

I am not sure on how to change TFormula::Compile for this. Or you mean adding a similar check like in this PR, but in:
TFormula::EvalParOld and TFormula::EvalParFast ?

Thks

Reply:
`TTreeFormula` inherits from `TFormula`.  Some of the steering code is in `TFormula` but the meat is in `TTreeFormula` (for parsing this include `DefinedVariable`, `ParseWithLeaf` etc.)

> Or you mean adding a similar check like in this PR, but in: TFormula::EvalParOld and TFormula::EvalParFast ?

This is not what I meant :) (and those are now mostly unused).

> I am not sure on how to change TFormula::Compile for this.

I suppose we need to understand 'why' `pos` gets out of bounds, see if it depends on the data (in which case I was wrong) or if it is structural (depends on the formula) in which case we need to (probably by stepping into Compile and  its inners functions) find how to detect that type of formula (in spirit this is similar the changes that you made to stop the 'formula is now too long because an alias' case).